client/vite.config.js

```diff
@@ -0,0 +1,106 @@
+import path from 'path';
+import react from '@vitejs/plugin-react';
+import strip from '@rollup/plugin-strip';
+import minifyTemplateLiterals from 'rollup-plugin-minify-html-literals';
+import { defineConfig } from 'vite';
+import { ViteFaviconsPlugin } from 'vite-plugin-favicon2';
+import { optimizeLodashImports } from '@optimize-lodash/rollup-plugin';
+import { taitoHtmlPlugin } from './src/html/html-plugin';
+
+const ANALYZE = !!process.env.ANALYZE;
+const OUT_DIR = path.resolve(__dirname, ANALYZE ? 'build' : '../../build');
+const DEV_HOST = process.env.DEV_BINDADDR || '127.0.0.1';
+const DEV_PORT = process.env.DEV_PORT || '3000';
+const PUBLIC_HOST = process.env.DOCKER_HOST ? '192.168.99.100' : 'localhost';
+const PUBLIC_PORT = process.env.COMMON_PUBLIC_PORT || DEV_PORT;
+
+export default defineConfig(({ mode }) => ({
+  publicDir: 'assets',
```

Comment:
To fix permission issues when using Linux and Docker we should add `cacheDir: '.vite'`. @hirvensaloa had edited the client Dockerfile in their projects to work around this but I'm not sure if that messes up with overlays and layers.